Makefile

```diff
@@ -1,7 +1,28 @@
 # Copyright (c) Facebook, Inc. and its affiliates. (http://www.facebook.com)
 
+define PYTHON_SCRIPT
+import os
+import sys
+sys.path.insert(0, os.path.abspath('../'))
+
+def skip(app, what, name, obj,would_skip, options):
+	if name in ( '__init__',):
+		return False
+	return would_skip
+
+	def setup(app):
+		app.connect('autodoc-skip-member', skip)
+	extensions.append('sphinx_autodoc_typehints')
+endef
+
 style:
 	black src/mapillary && flake8 src/mapillary
 
+docs:
+	sphinx-apidoc -o sphinx-docs ./src/ sphinx-apidoc --full -A 'Mapillary'; cd sphinx-docs; echo "$$PYTHON_SCRIPT" >> conf.py; make markdown;
+
+docs-clean:
+	rm -rf docs/
+
```

Comment:
Can you explain the changes in this file?

Reply:
As per the answer by the OP at this [StackOverflow post](https://stackoverflow.com/questions/36237477/python-docstrings-to-github-readme-md), a step included for performing type-hint checking in the documentation was to append a piece of Python code.

Unfortunately, I did not find a clean and easy way of dealing with that as one line in the Makefile (because `cd` cannot exist on separate lines). A solution to this was to create a variable with the above syntax, and just append that to the file whenever new `sphinx-docs` are generated.

As for the `docs` target,

1. `sphinx-apidoc -o sphinx-docs ./src/ sphinx-apidoc --full -A 'Mapillary'` uses `sphinx` API documentation generation (from the [SO post](https://stackoverflow.com/questions/36237477/python-docstrings-to-github-readme-md)), and outputs the `sphinx-docs` folder by using the `./src/` directory as the root of the DocStrings. The `--full` flag includes all default options, and `-A 'Mapillary'`, sets the name of the user generating the documentation to `Mapillary`.
2. Makefile does not allow `cd` as expected. If I cd on different lines, it will move back to the origin of the Makefile in the next line, thus, I have to 'chain' these commands together.
3. Now that I'm in `sphinx-docs`, I need to append to conf.py the python code from the SO post
4. Then I need to generate the markdown from the previously generated artifact `.rst` files